File: examples/hello-world/gabriel-greeting-go/api/public_test.go
package api_test

import (
	"testing"

	"gabriel-greeting-go/api"
	pb "gabriel-greeting-go/gen/go/greeting/v1"
)

func TestListLanguagesIncludesEnglish(t *testing.T) {
	response, err := api.ListLanguages()
	if err != nil {
		t.Fatalf("ListLanguages() error = %v", err)
	}
	if len(response.GetLanguages()) == 0 {
		t.Fatal("ListLanguages() returned no languages")
	}

	for _, language := range response.GetLanguages() {
		if language.GetCode() == "en" {
			if language.GetName() != "English" {
				t.Fatalf("English name = %q, want %q", language.GetName(), "English")
			}
			if language.GetNative() != "English" {
				t.Fatalf("English native = %q, want %q", language.GetNative(), "English")
			}
			return
		}
	}

	t.Fatal(`ListLanguages() did not include language code "en"`)
}

func TestSayHelloUsesRequestedLanguage(t *testing.T) {
	response, err := api.SayHello(&pb.SayHelloRequest{
		Name:     "Alice",
		LangCode: "fr",
	})
	if err != nil {
		t.Fatalf("SayHello() error = %v", err)
	}
	if response.GetGreeting() != "Bonjour Alice" {
		t.Fatalf("Greeting = %q, want %q", response.GetGreeting(), "Bonjour Alice")
	}
	if response.GetLanguage() != "French" {
		t.Fatalf("Language = %q, want %q", response.GetLanguage(), "French")
	}
	if response.GetLangCode() != "fr" {
		t.Fatalf("LangCode = %q, want %q", response.GetLangCode(), "fr")
	}
}

func TestSayHelloFallsBackToEnglishDefaultName(t *testing.T) {
	response, err := api.SayHello(&pb.SayHelloRequest{
		LangCode: "unknown",
	})
	if err != nil {
		t.Fatalf("SayHello() error = %v", err)
	}
	if response.GetGreeting() != "Hello Mary" {
		t.Fatalf("Greeting = %q, want %q", response.GetGreeting(), "Hello Mary")
	}
	if response.GetLanguage() != "English" {
		t.Fatalf("Language = %q, want %q", response.GetLanguage(), "English")
	}
	if response.GetLangCode() != "en" {
		t.Fatalf("LangCode = %q, want %q", response.GetLangCode(), "en")
	}
}

File: examples/hello-world/gabriel-greeting-rust/api/public_test.rs
use crate::gen::rust::greeting::v1 as pb;

#[test]
fn list_languages_includes_english() {
    let response = crate::public::list_languages(pb::ListLanguagesRequest::default());

    assert!(!response.languages.is_empty());
    let english = response
        .languages
        .iter()
        .find(|language| language.code == "en")
        .expect("English should be present");

    assert_eq!(english.name, "English");
    assert_eq!(english.native, "English");
}

#[test]
fn say_hello_uses_requested_language() {
    let response = crate::public::say_hello(pb::SayHelloRequest {
        name: "Alice".to_string(),
        lang_code: "fr".to_string(),
    });

    assert_eq!(response.greeting, "Bonjour Alice");
    assert_eq!(response.language, "French");
    assert_eq!(response.lang_code, "fr");
}

#[test]
fn say_hello_uses_localized_default_name() {
    let response = crate::public::say_hello(pb::SayHelloRequest {
        name: String::new(),
        lang_code: "ja".to_string(),
    });

    assert_eq!(response.greeting, "こんにちは、マリアさん");
    assert_eq!(response.language, "Japanese");
    assert_eq!(response.lang_code, "ja");
}

#[test]
fn say_hello_falls_back_to_english_for_unknown_language() {
    let response = crate::public::say_hello(pb::SayHelloRequest {
        name: String::new(),
        lang_code: "unknown".to_string(),
    });

    assert_eq!(response.greeting, "Hello Mary");
    assert_eq!(response.language, "English");
    assert_eq!(response.lang_code, "en");
}
